### PR TITLE
Test adding different newsletters into the onboarding flow

### PR DESCRIPTION
**Pull request by liamness**
## What does this change?

This sets up an A/B test where 50% of users will see different newsletter options when going through the onboarding flow.

## How to test

Set a `GU_mvt_id` cookie to `1` to see the changes being tested. Set it to `2` to see the control group behaviour, which shows the same list as is currently used.
Still looking into how best to test region-specific behaviour.

## Accessibility

- [ ] [Tested with screen reader](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#screen-readers)
- [ ] [Navigable with keyboard](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#keyboard-navigation)
- [ ] [Colour contrast passed](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#colour-contrast)
- [ ] [The change doesn't use only colour to convey meaning](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#use-of-colour)

### CODE REVIEW

File: src/server/routes/consents.ts [hunk 1]
```diff
@@ -279,9 +294,23 @@ router.get(
...
       pageTitle = _pageTitle;
 
+      // TODO: decide which approach to use
+      const newslettersTestActive =
+        state.abTesting.participations[ONBOARDING_NEWSLETTERS_TEST_ID]
+          ?.variant === ONBOARD_NEWSLETTERS_TEST_VARIANT;
+      // const newslettersTestActive = state.abTestAPI.isUserInVariant(
+      //   ONBOARDING_NEWSLETTERS_TEST_ID,
+      //   ONBOARD_NEWSLETTERS_TEST_VARIANT,
+      // );
```
Comment (coldlink) [inline src/server/routes/consents.ts:304]: in general this is the approach that we've taken in the past

File: src/server/routes/consents.ts [hunk 2]
```diff
@@ -229,7 +244,7 @@ export const consentPages: ConsentPage[] = [
         page: 'review',
         consents: await getUserConsentsForPage(ALL_CONSENT, ip, sc_gu_u),
         newsletters: await getUserNewsletterSubscriptions(
-          NewsletterMap.get(geo) as string[],
+          TestNewsletterMap.get(geo) as string[],
```
Comment (coldlink) [inline src/server/routes/consents.ts:247]: This change seems to fall outside the test, which may cause issues with reading newsletter subscriptions

File: src/server/routes/consents.ts (hunk 1 above)
Comment (liamness) [reply src/server/routes/consents.ts]: Okay, will stick with that then.

Comment (liamness) [reply src/server/routes/consents.ts]: (discussed away from this PR with Mahesh)

For some reason `abTestAPI` was not available during cypress runs, so have switched to the other approach to unblock this feature.

**Pull request closed without merging**